main/streams/userspace.c

```diff
@@ -1389,6 +1397,53 @@ static int user_wrapper_stat_url(php_stream_wrapper *wrapper, const char *url, i
 	return ret;
 
 }
+static int user_wrapper_is_cacheable(php_stream_wrapper *wrapper, const char *url, int flags, php_stream_context *context)
+{
+	struct php_user_stream_wrapper *uwrap = (struct php_user_stream_wrapper*)wrapper->abstract;
+	zval zfuncname, zretval;
+	zval args[2];
+	int call_result;
+	zval object;
+	int ret = -1;
+
+	/* create an instance of our class */
+	user_stream_create_object(uwrap, context, &object);
+	if (Z_TYPE(object) == IS_UNDEF) {
+		return ret;
+	}
+
+	/* call it's is_cacheable method - set up params first */
+	ZVAL_STRING(&args[0], url);
+	ZVAL_LONG(&args[1], flags);
+
+	ZVAL_STRING(&zfuncname, USERSTREAM_IS_CACHEABLE);
+
+	call_result = call_user_function_ex(NULL,
+			&object,
+			&zfuncname,
+			&zretval,
+			2, args,
+			0, NULL	);
+
+	if (call_result == SUCCESS) {
+		/* We got the info we needed */
+		convert_to_long(&zretval);
```

Comment:
Why long? You just documented it as being true/false. Then you should use zend_is_true or something like that.